src/transformers/tokenization_utils_base.py

```diff
@@ -2456,8 +2456,10 @@ def save_pretrained(
 
         # Add tokenizer class to the tokenizer config to be able to reload it with from_pretrained
         tokenizer_class = self.__class__.__name__
-        # Remove the Fast at the end unless we have a special `PreTrainedTokenizerFast`
-        if tokenizer_class.endswith("Fast") and tokenizer_class != "PreTrainedTokenizerFast":
+        # Remove the Fast at the end if we can save the slow tokenizer
+        if tokenizer_class.endswith("Fast") and (
+            hasattr(self, "can_save_slow_tokenizer") and self.can_save_slow_tokenizer
```

Comment:
You can have this check shorter with `getattr`:
```suggestion
            getattr(self, "can_save_slow_tokenizer", False)
```